src/System.IO.Pipelines.Text.Primitives/ReadableBufferExtensions.cs

```diff
@@ -197,27 +189,17 @@ public unsafe static string GetAsciiString(this ReadableBuffer buffer)
 
                 foreach (var memory in buffer)
                 {
-                    void* pointer;
-                    if (memory.TryGetPointer(out pointer))
+                    var handle = memory.GetPinnedMemoryHandle();
+                    try
                     {
-                        if (!AsciiUtilities.TryGetAsciiString((byte*)pointer, output + offset, memory.Length))
+                        if (!AsciiUtilities.TryGetAsciiString((byte*)handle.PinnedPointer, output + offset, memory.Length))
                         {
                             throw new InvalidOperationException();
                         }
                     }
-                    else
+                    finally
                     {
-                        ArraySegment<byte> data;
-                        if (memory.TryGetArray(out data))
-                        {
-                            fixed (byte* ptr = &data.Array[0])
```

Comment:
Using the new API, fixed has changed into `GCHandle` pinning. Is this acceptable?